src/quantum/azext_quantum/operations/job.py

```diff
@@ -34,12 +34,31 @@ def get(cmd, job_id, resource_group_name=None, workspace_name=None, location=Non
     return client.get(job_id)
 
 
+def _check_dotnet_available():
+    """
+    Will fail if dotnet cannot be executed on the system.
+    """
+    args = ["dotnet", "--version"]
+
+    try:
+        import subprocess
+        result = subprocess.run(args, stdout=subprocess.PIPE, check=False)
+    except FileNotFoundError:
+        raise CLIError(f"Could not find 'dotnet' on the system.")
+
+    if result.returncode != 0:
+        raise CLIError(f"Failed to run 'dotnet'. (Error {result.returncode})")
```

Comment:
We are going the deprecate the usage of `CLIError` type. Could you following the [doc](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md) to use a more specific error type.

Reply:
Thanks @kairu-ms . I've opened the following issue to track this: #3805